pycloudmessenger/ffl/fflapi.py

```diff
@@ -169,8 +171,28 @@ def task_assignment_update(self, task_name: str, status: str, model: dict = None
         Throws: An exception on failure
         Returns: Nothing
         '''
-        message = self.catalog.msg_task_assignment_update(task_name, status,
-                                                          model, want_reply=False)
+
+        if model is not None:
+            message = self.catalog.msg_bin_uploader()
+            upload_info = self._invoke_service(message)
+            response = requests.post(
+                upload_info['url'],
+                files={'file': str(model)},
+                data=upload_info['fields'],
+                headers=None)
+
+            if not response.ok:
+                raise Exception('response: %d' % response.status_code)
+
+            message = self.catalog.msg_bin_downloader(upload_info['fields']['key'])
+            download_info = self._invoke_service(message)
+
+        message = self.catalog.msg_task_assignment_update(
+            task_name,
+            status,
+            {'url': download_info},
```

Comment:
download_info could be unknown here if no model

Reply:
added a default `msg = {}`

Reply:
??